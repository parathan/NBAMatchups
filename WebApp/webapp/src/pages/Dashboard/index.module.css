.dashboard {
    padding: 10vh;
    text-align: center;
    animation: fade-in 2s;
}

.header {
    color: var(--header-font-colour);
    font-size: calc(10px + 3vmin);;
    padding-bottom: 2vh;
    margin-top: 8%;
}

.hidden {
    visibility: hidden;
    height: 0;
}

.notHidden {
    visibility: visible;
}

.input {
    padding-top: 5vh;
}

.dropdown {
    text-align: center;
    padding-left: 2vw;
    padding-right: 2vw;
    background-color: var(--secondary-colour);
    width: 20em;
    height: 3em;
    border-radius: .25em;
    overflow: hidden;
}

.dropdown:hover {
    opacity: .75;
}


@keyframes fade-in {
    from {
        opacity: 0;
    }
    to {
        opacity: 100;
    }
}