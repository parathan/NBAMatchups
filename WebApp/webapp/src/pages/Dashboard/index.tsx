import { ChangeEvent, useEffect, useState } from 'react';
import styles from './index.module.css';
import Layout from "../../components/Layout/Layout";
import { Alert, CircularProgress, Grid } from '@mui/material';

import axios from 'axios';
import { TotalTeamData, TeamData } from '../../interfaces/TotalTeamData';
import { ChartFormat } from '../../interfaces/ChartFormat';
import { teamsNames } from '../../constants/teamNames';
import { statsArray } from '../../constants/statArray';

import {
    Chart as ChartJS,
    CategoryScale,
    LinearScale,
    PointElement,
    LineElement,
    Title,
    Tooltip,
    Legend,
  } from 'chart.js';
  import { Line } from 'react-chartjs-2';
import { statsMap } from '../../constants/statDictionary';
import { allTeams, allTeamsCached } from '../../constants/routes';
import colours from '../../constants/colours';

const testData: ChartFormat = {
    labels: ["2019", "2020", "2021", "2022", "2023"],
    datasets: [
        {
            label: "Data",
            backgroundColor: colours.RED, // Setting up the background color for the dataset
            borderColor: colours.RED,
            data: []
        },
        {
            label: "League Average",
            backgroundColor: colours.BLUE, // Setting up the background color for the dataset
            borderColor: colours.BLUE,
            data: []
        }
    ]
}

ChartJS.register(
    CategoryScale,
    LinearScale,
    PointElement,
    LineElement,
    Title,
    Tooltip,
    Legend
);

const options = {
    responsive: true,
    plugins: {
        legend: {
            position: 'top' as const,
            labels: {
                color: colours.P_FONT_COLOUR,
                font: {
                    size: 18
                }
            }
        },
        title: {
            display: false,
        }
    },
    scales: {
        x: {
            ticks: {
                color: colours.P_FONT_COLOUR,
                font: {
                    size: 15
                }
            },
        },
        y: {
            ticks: {
                color: colours.P_FONT_COLOUR,
                font: {
                    size: 15
                }
            },
        }
    },
};


/**
 * This Dashboard page is used to show the trends of a team's stats over the past few years, and compare them to the league average for that 
 * stat over the same time period. It take an API call from the /teams/allTeams route to get the data, and use the ChartJS library to
 * display this data.
 * @returns 
 */
function Dashboard() {

    const [data, setData] = useState<TotalTeamData[]>([]) // Total Data to be fetched
    const [meanData, setMeanData] = useState<TotalTeamData>() // Mean Data will be saved seperately to be accessed more efficiently
    const [progress, setProgress] = useState(true) // For progress bar
    const [error, setError] = useState(false) // For Error message if error fetching data
    const [errMessage, setErrorMessage] = useState("Error")
    // TODO #5
    const startYear: number = 2019;
    const endYear: number = 2023;
    const years: string[] = ["2019","2020","2021","2022","2023"]
    const [team, setTeam] = useState("") // The team name who's data will be displayed
    const [field, setField] = useState("") // The team's field name that will be displayed
    const [chartData, setChartData] = useState<ChartFormat>(testData)

    // Intended that useEffect runs once, as empty array given means that it only updates once when page renders.
    // In development, it will run twice due to strict mode being on. This won't happen in production, however.
    // https://byby.dev/useeffect-run-twice
    useEffect(() => {
        // Gets data from route that uses redis cache. If it fails it uses route that doesnt have cache
        const getCachedData = async () => {
            axios.post(process.env.REACT_APP_API_URL + allTeamsCached, {
                startYear: startYear,
                endYear: endYear
            })
            .then((response) => {
                setData(response.data)
                setProgress(false)
    
                let meanData: TotalTeamData | undefined = response.data.find((givenTeam: { teamName: string; }) => {
                    return givenTeam.teamName === "MEAN"
                })
                setMeanData(meanData)
            })
            .catch((error) => {
                console.log(error)
                getUncachedData();
            })
        }

        const getUncachedData = async () => {
            axios.post(process.env.REACT_APP_API_URL + allTeams, {
                startYear: startYear,
                endYear: endYear
            })
            .then((response) => {
                setData(response.data)
                setProgress(false)
    
                let meanData: TotalTeamData | undefined = response.data.find((givenTeam: { teamName: string; }) => {
                    return givenTeam.teamName === "MEAN"
                })
                setMeanData(meanData)
            })
            .catch((error) => {
                setProgress(false)
                setError(true)
                let errorMessage = "Error retrieving from Server\n"
                setErrorMessage(errorMessage.concat(error))
            })
        }

        getCachedData();
        
    }, []) // empty array so this only updates once on render

    function establishData(team: string, field: string) {
        let labelName = statsMap.get(field) || ""
        let teamData: TotalTeamData | undefined = data.find(givenTeam => {
            return givenTeam.teamName === team
        })

        let fieldData: number[] = []
        teamData?.stats.forEach(function (yearlyStat) {
            fieldData.push(Number(yearlyStat.yearStats[field as keyof TeamData]))
            // setDisplayData(oldArray => [...oldArray, Number(yearlyStat.yearStats[field as keyof TeamData])])
        })

        let averageData: number[] = []
        meanData?.stats.forEach(function (yearlyStat) {
            averageData.push(Number(yearlyStat.yearStats[field as keyof TeamData]))
            // setDisplayData(oldArray => [...oldArray, Number(yearlyStat.yearStats[field as keyof TeamData])])
        })

        let newChart: ChartFormat = {
            labels: years,
            datasets: [
                {
                    label: labelName,
                    backgroundColor: "rgb(255, 99, 132)", // Setting up the background color for the dataset
                    borderColor: "rgb(255, 99, 132)",
                    data: fieldData
                },
                {
                    label: "League Average",
                    backgroundColor: "rgb(53, 162, 235)", // Setting up the background color for the dataset
                    borderColor: "rgba(53, 162, 235, 0.5)",
                    data: averageData
                }
            ]
        }
        setChartData(newChart)
        console.log("Both")
    }

    function changeTeam(e: ChangeEvent<HTMLSelectElement>) {
        let team = e.target.value // needs to be seperate as set functions are async, so can't use them in function
        setTeam(team)
        if (field !== "") {
            establishData(team, field)
        }
    }
    
    function changeField(e: ChangeEvent<HTMLSelectElement>) {
        let field = e.target.value // needs to be seperate as set functions are async, so can't use them in function
        setField(field)
        if (team !== "") {
            establishData(team, field)
        }
    }

    return (
        <Layout>
            <div className={styles.dashboard}>
                <div className={styles.header}>Dashboard</div>
                {progress ? <CircularProgress /> : null}
                {
                    error ? 
                    <div className={styles.errMessage}>
                        <Alert severity='error'>
                            {errMessage}
                        </Alert>
                    </div>
                    : 
                    null}
                {
                    error || progress ? 
                    null
                    :
                    <div>
                        <Grid container spacing={2} className={styles.input}>
                            <Grid item xs={2}></Grid>
                            <Grid item xs={4}>
                            <select onChange={changeTeam} className={styles.dropdown}>
                                <option key={""} value="">Pick a Team</option>
                                {teamsNames.map( teamName =>
                                    <option data-testid="team-options" key={teamName} value={teamName}>{teamName}</option>
                                )};
                            </select>
                            </Grid>
                            <Grid item xs={4}>
                            <select onChange={changeField} className={styles.dropdown}>
                                <option key={""} value="">Pick a Field</option>
                                {statsArray.map( stat =>
                                    <option data-testid="field-options" key={stat[0]} value={stat[0]}>{stat[1]}</option>
                                )}
                            </select>
                            </Grid>
                            <Grid item xs={2}></Grid>
                            <Grid item xs={3}></Grid>
                            <Grid item xs={6}>
                                <Line options={options} data={chartData}/>
                            </Grid>
                            <Grid item xs={3}></Grid>
                        </Grid>
                    </div>
                }
                
            </div>
        </Layout>
    )
}

export default Dashboard;