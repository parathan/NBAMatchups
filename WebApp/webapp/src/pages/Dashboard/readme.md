# Dashboard

## Description

The Dashboard component is a React component designed to display statistical data for different teams over multiple years. It allows users to select a team and a specific statistical field to visualize data using a line chart. The line chart then displays the data for that team as well as the league average for that statistic over the given years.

## Props
This component does not accept any props.

## State
**data**: An array of TotalTeamData objects fetched from the API containing statistical data for all teams.  
**meanData**: A TotalTeamData object containing the mean statistical data across all teams taken from the API response  
**progress**: A boolean value indicating whether data is being fetched from the API (true) or not (false).  
**error**: A boolean value indicating whether an error occurred while fetching data from the API (true) or not (false).  
**team**: The name of the selected team for which data will be displayed.  
**field**: The selected statistical field for which data will be displayed.  
**chartData**: A ChartFormat object containing data for the line chart.  

## Functions
**establishData(team: string, field: string)**: Function to establish data for the selected team and field. It updates the chartData state.  
**changeTeam(e: ChangeEvent<HTMLSelectElement>)**: Event handler function to change the selected team. It triggers the establishment of data for the selected team and field.  
**changeField(e: ChangeEvent<HTMLSelectElement>)**: Event handler function to change the selected field. It triggers the establishment of data for the selected team and field.  

## Dependencies
**axios**: For making HTTP requests to fetch data from the API.  
**react-chartjs-2**: React wrapper for Chart.js library to render line charts.  

## External Resources
**Chart.js**: A JavaScript library for creating charts, used for rendering the line chart.  
**Material-UI**: A popular React UI framework used for styling and layout components.  

## Styles
The component uses CSS modules for styling. The main styles are defined in the index.module.css file.

## Notes
The component fetches data from an API endpoint to populate the statistical information.  
It handles loading and error states to provide a better user experience.  
Users can select a team and a specific statistical field to visualize data using the line chart.  
