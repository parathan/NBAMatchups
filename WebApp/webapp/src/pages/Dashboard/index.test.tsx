import renderer from 'react-test-renderer'
import Dashboard from '.'
import { BrowserRouter } from 'react-router-dom'


it('Dashboard renders', () => {
    const component = renderer.create(
        <BrowserRouter>
            <Dashboard/>
        </BrowserRouter>,
    )
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})