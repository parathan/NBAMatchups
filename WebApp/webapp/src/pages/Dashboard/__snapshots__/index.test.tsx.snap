// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard renders 1`] = `
<div
  className="layout"
>
  <div
    className="navbar"
  >
    <aside
      className="ps-sidebar-root css-17c6mwq"
      data-testid="ps-sidebar-root-test-id"
      width="250px"
    >
      <div
        className="ps-sidebar-container css-dip3t8"
        data-testid="ps-sidebar-container-test-id"
      >
        <nav
          className="ps-menu-root css-vj11vy"
        >
          <ul
            className="css-ewdv3l"
          >
            <li
              className="ps-menuitem-root css-1t8x7v1"
              disabled={false}
            >
              <a
                className="ps-menu-button"
                data-testid="ps-menu-button-test-id"
                href="/"
                onClick={[Function]}
                tabIndex={0}
              >
                <span
                  className="ps-menu-label css-12w9als"
                >
                  Home
                </span>
              </a>
            </li>
            <li
              className="ps-menuitem-root css-1t8x7v1"
              disabled={false}
            >
              <a
                className="ps-menu-button"
                data-testid="ps-menu-button-test-id"
                href="/matchups"
                onClick={[Function]}
                tabIndex={0}
              >
                <span
                  className="ps-menu-label css-12w9als"
                >
                  Team Matchups
                </span>
              </a>
            </li>
            <li
              className="ps-menuitem-root css-1t8x7v1"
              disabled={false}
            >
              <a
                className="ps-menu-button"
                data-testid="ps-menu-button-test-id"
                href="/predictions"
                onClick={[Function]}
                tabIndex={0}
              >
                <span
                  className="ps-menu-label css-12w9als"
                >
                  Win/Loss Prediction
                </span>
              </a>
            </li>
            <li
              className="ps-menuitem-root css-1t8x7v1"
              disabled={false}
            >
              <a
                className="ps-menu-button"
                data-testid="ps-menu-button-test-id"
                href="/dashboard"
                onClick={[Function]}
                tabIndex={0}
              >
                <span
                  className="ps-menu-label css-12w9als"
                >
                  Dashboard
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </aside>
  </div>
  <div
    className="children"
  >
    <div
      className="dashboard"
    >
      <div
        className="header"
      >
        Dashboard
      </div>
      <div
        className="notHidden"
      >
        <span
          className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
          role="progressbar"
          style={
            {
              "height": 40,
              "width": 40,
            }
          }
        >
          <svg
            className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
              cx={44}
              cy={44}
              fill="none"
              r={20.2}
              strokeWidth={3.6}
              style={{}}
            />
          </svg>
        </span>
      </div>
      <div
        className="hidden"
      >
        Errors getting data
      </div>
      <div
        className="hidden"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 input css-mhc70k-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
          />
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
          >
            <select
              className="dropdown"
              onChange={[Function]}
            >
              <option
                value=""
              >
                Pick a Team
              </option>
              <option
                value="Atlanta Hawks"
              >
                Atlanta Hawks
              </option>
              <option
                value="Boston Celtics"
              >
                Boston Celtics
              </option>
              <option
                value="Brooklyn Nets"
              >
                Brooklyn Nets
              </option>
              <option
                value="Charlotte Hornets"
              >
                Charlotte Hornets
              </option>
              <option
                value="Chicago Bulls"
              >
                Chicago Bulls
              </option>
              <option
                value="Cleveland Cavaliers"
              >
                Cleveland Cavaliers
              </option>
              <option
                value="Dallas Mavericks"
              >
                Dallas Mavericks
              </option>
              <option
                value="Denver Nuggets"
              >
                Denver Nuggets
              </option>
              <option
                value="Detroit Pistons"
              >
                Detroit Pistons
              </option>
              <option
                value="Golden State Warriors"
              >
                Golden State Warriors
              </option>
              <option
                value="Houston Rockets"
              >
                Houston Rockets
              </option>
              <option
                value="Indiana Pacers"
              >
                Indiana Pacers
              </option>
              <option
                value="Los Angeles Clippers"
              >
                Los Angeles Clippers
              </option>
              <option
                value="Los Angeles Lakers"
              >
                Los Angeles Lakers
              </option>
              <option
                value="Memphis Grizzlies"
              >
                Memphis Grizzlies
              </option>
              <option
                value="Miami Heat"
              >
                Miami Heat
              </option>
              <option
                value="Milwaukee Bucks"
              >
                Milwaukee Bucks
              </option>
              <option
                value="Minnesota Timberwolves"
              >
                Minnesota Timberwolves
              </option>
              <option
                value="New Orleans Pelicans"
              >
                New Orleans Pelicans
              </option>
              <option
                value="New York Knicks"
              >
                New York Knicks
              </option>
              <option
                value="Oklahoma City Thunder"
              >
                Oklahoma City Thunder
              </option>
              <option
                value="Orlando Magic"
              >
                Orlando Magic
              </option>
              <option
                value="Philadelphia 76ers"
              >
                Philadelphia 76ers
              </option>
              <option
                value="Phoenix Suns"
              >
                Phoenix Suns
              </option>
              <option
                value="Portland Trail Blazers"
              >
                Portland Trail Blazers
              </option>
              <option
                value="Sacramento Kings"
              >
                Sacramento Kings
              </option>
              <option
                value="San Antonio Spurs"
              >
                San Antonio Spurs
              </option>
              <option
                value="Toronto Raptors"
              >
                Toronto Raptors
              </option>
              <option
                value="Utah Jazz"
              >
                Utah Jazz
              </option>
              <option
                value="Washington Wizards"
              >
                Washington Wizards
              </option>
              ;
            </select>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
          >
            <select
              className="dropdown"
              onChange={[Function]}
            >
              <option
                value=""
              >
                Pick a Field
              </option>
              <option
                value="g"
              >
                Games Played
              </option>
              <option
                value="mp"
              >
                Minutes Played
              </option>
              <option
                value="fg"
              >
                Field Goals
              </option>
              <option
                value="fga"
              >
                Field Goals Attempted
              </option>
              <option
                value="fg_pct"
              >
                Field Goal Percentage
              </option>
              <option
                value="fg3"
              >
                3-Point Field Goals
              </option>
              <option
                value="fg3a"
              >
                3-Point Field Goals Attempted
              </option>
              <option
                value="fg3_pct"
              >
                3-Point Field Goal Percentage
              </option>
              <option
                value="fg2"
              >
                2-Point Field Goals
              </option>
              <option
                value="fg2a"
              >
                2-Point Field goals Attempted
              </option>
              <option
                value="fg2_pct"
              >
                2-Point Field Goal Percentage
              </option>
              <option
                value="ft"
              >
                Free Throws
              </option>
              <option
                value="fta"
              >
                Free Throws Attempted
              </option>
              <option
                value="ft_pct"
              >
                Free Throw Percentage
              </option>
              <option
                value="orb"
              >
                Offensive Rebounds
              </option>
              <option
                value="drb"
              >
                Defensive Rebounds
              </option>
              <option
                value="trb"
              >
                Total Rebounds
              </option>
              <option
                value="ast"
              >
                Assists
              </option>
              <option
                value="stl"
              >
                Steals
              </option>
              <option
                value="blk"
              >
                Blocks
              </option>
              <option
                value="tov"
              >
                Turnovers
              </option>
              <option
                value="pf"
              >
                Personal Fouls
              </option>
              <option
                value="pts"
              >
                Points
              </option>
              <option
                value="opp_fg"
              >
                Opponent Field Goals
              </option>
              <option
                value="opp_fga"
              >
                Opponent Field Goals Attempted
              </option>
              <option
                value="opp_fg_pct"
              >
                Opponent Field Goal Percentage
              </option>
              <option
                value="opp_fg3"
              >
                Opponent 3-Point Field Goals
              </option>
              <option
                value="opp_fg3a"
              >
                Opponent 3-Point Field Goals Attempted
              </option>
              <option
                value="opp_fg3_pct"
              >
                Opponent 3-Point Field Goal Percentage
              </option>
              <option
                value="opp_fg2"
              >
                Opponent 2-Point Field Goals
              </option>
              <option
                value="opp_fg2a"
              >
                Opponent 2-Point Field goals Attempted
              </option>
              <option
                value="opp_fg2_pct"
              >
                Opponent 2-Point Field Goal Percentage
              </option>
              <option
                value="opp_ft"
              >
                Opponent Free Throws
              </option>
              <option
                value="opp_fta"
              >
                Opponent Free Throws Attempted
              </option>
              <option
                value="opp_ft_pct"
              >
                Opponent Free Throw Percentage
              </option>
              <option
                value="opp_orb"
              >
                Opponent Offensive Rebounds
              </option>
              <option
                value="opp_drb"
              >
                Opponent Defensive Rebounds
              </option>
              <option
                value="opp_trb"
              >
                Opponent Total Rebounds
              </option>
              <option
                value="opp_ast"
              >
                Opponent Assists
              </option>
              <option
                value="opp_stl"
              >
                Opponent Steals
              </option>
              <option
                value="opp_blk"
              >
                Opponent Blocks
              </option>
              <option
                value="opp_tov"
              >
                Opponent Turnovers
              </option>
              <option
                value="opp_pf"
              >
                Opponent Personal Fouls
              </option>
              <option
                value="opp_pts"
              >
                Opponent Points
              </option>
              <option
                value="age"
              >
                Age
              </option>
              <option
                value="wins"
              >
                Wins
              </option>
              <option
                value="losses"
              >
                Losses
              </option>
              <option
                value="wins_pyth"
              >
                Pythagorean Wins
              </option>
              <option
                value="losses_pyth"
              >
                Pythagorean Losses
              </option>
              <option
                value="mov"
              >
                Margin of Victory
              </option>
              <option
                value="sos"
              >
                Strength of Schedule
              </option>
              <option
                value="srs"
              >
                Simple Rating System
              </option>
              <option
                value="off_rtg"
              >
                Offensive Rating
              </option>
              <option
                value="def_rtg"
              >
                Defensive Rating
              </option>
              <option
                value="net_rtg"
              >
                Net Rating
              </option>
              <option
                value="pace"
              >
                Pace
              </option>
              <option
                value="fta_per_fga_pct"
              >
                Free Throw Attempt Rate
              </option>
              <option
                value="fg3a_per_fga_pct"
              >
                3-Point Attempt Rate
              </option>
              <option
                value="ts_pct"
              >
                True Shooting Percentage
              </option>
              <option
                value="efg_pct"
              >
                Effective Field Goal Percentage
              </option>
              <option
                value="tov_pct"
              >
                Turnover Percentage
              </option>
              <option
                value="orb_pct"
              >
                Offensive Rebound Percentage
              </option>
              <option
                value="ft_rate"
              >
                Free Throws Per Field Goal Attempt
              </option>
              <option
                value="opp_efg_pct"
              >
                Opponent Effective Field Goal Percentage
              </option>
              <option
                value="opp_tov_pct"
              >
                Opponent Turnover Percentage
              </option>
              <option
                value="drb_pct"
              >
                Defensive Rebound Percentage
              </option>
              <option
                value="opp_ft_rate"
              >
                Opponent Free throws Per Field Goal Attempt
              </option>
              <option
                value="avg_dist"
              >
                Average Distance of FGA
              </option>
              <option
                value="pct_fga_fg2a"
              >
                Percentage of FGA that are 2-Pt FGAs
              </option>
              <option
                value="pct_fga_00_03"
              >
                Percentage of FGA that are 0-3 feet from basket
              </option>
              <option
                value="pct_fga_03_10"
              >
                Percentage of FGA that are 3-10 feet from basket
              </option>
              <option
                value="pct_fga_10_16"
              >
                Percentage of FGA that are 10-16 feet from basket
              </option>
              <option
                value="pct_fga_16_xx"
              >
                Percentage of 2-Pt FGA that are 16+ feet from the basket
              </option>
              <option
                value="pct_fga_fg3a"
              >
                Percentage of FGA that are 3-Pt FGAS
              </option>
              <option
                value="fg_pct_00_03"
              >
                FG Percentage of FGA that are 0-3 feet from basket
              </option>
              <option
                value="fg_pct_03_10"
              >
                FG Percentage of FGA that are 3-10 feet from basket
              </option>
              <option
                value="fg_pct_10_16"
              >
                FG Percentage of FGA that are 10-16 feet from basket
              </option>
              <option
                value="fg_pct_16_xx"
              >
                FG Percentage of 2-Pt FGA that are 16+ feet from basket
              </option>
              <option
                value="pct_ast_fg2"
              >
                Percentage of 2-Pt FG assisted on
              </option>
              <option
                value="pct_ast_fg3"
              >
                Percentage of 3-Pt FG assisted on
              </option>
              <option
                value="pct_fga_dunk"
              >
                Percentage of FGA that are dunks
              </option>
              <option
                value="fg_dunk"
              >
                Number of made dunks
              </option>
              <option
                value="pct_fga_layup"
              >
                Percentage of FGA that are layups
              </option>
              <option
                value="fg_layup"
              >
                Number of made layups
              </option>
              <option
                value="pct_fg3a_corner"
              >
                Percentage of 3-Pt attempts from the corner
              </option>
              <option
                value="fg3_pct_corner"
              >
                Corner 3-Pt Percentage
              </option>
              <option
                value="opp_avg_dist"
              >
                Opponent Average Distance of FGA
              </option>
              <option
                value="opp_pct_fga_fg2a"
              >
                Opponent Percentage of FGA that are 2-Pt FGAs
              </option>
              <option
                value="opp_pct_fga_00_03"
              >
                Opponent Percentage of FGA that are 0-3 feet from basket
              </option>
              <option
                value="opp_pct_fga_03_10"
              >
                Opponent Percentage of FGA that are 3-10 feet from basket
              </option>
              <option
                value="opp_pct_fga_10_16"
              >
                Opponent Percentage of FGA that are 10-16 feet from basket
              </option>
              <option
                value="opp_pct_fga_16_xx"
              >
                Opponent Percentage of 2-Pt FGA that are 16+ feet from the basket
              </option>
              <option
                value="opp_pct_fga_fg3a"
              >
                Opponent Percentage of FGA that are 3-Pt FGAS
              </option>
              <option
                value="opp_fg_pct_00_03"
              >
                Opponent FG Percentage of FGA that are 0-3 feet from basket
              </option>
              <option
                value="opp_fg_pct_03_10"
              >
                Opponent FG Percentage of FGA that are 3-10 feet from basket
              </option>
              <option
                value="opp_fg_pct_10_16"
              >
                Opponent FG Percentage of FGA that are 10-16 feet from basket
              </option>
              <option
                value="opp_fg_pct_16_xx"
              >
                Opponent FG Percentage of 2-Pt FGA that are 16+ feet from basket
              </option>
              <option
                value="opp_pct_ast_fg2"
              >
                Opponent Percentage of 2-Pt FG assisted on
              </option>
              <option
                value="opp_pct_ast_fg3"
              >
                Opponent Percentage of 3-Pt FG assisted on
              </option>
              <option
                value="opp_pct_fga_dunk"
              >
                Opponent Percentage of FGA that are dunks
              </option>
              <option
                value="opp_fg_dunk"
              >
                Opponent Number of made dunks
              </option>
              <option
                value="opp_pct_fga_layup"
              >
                Opponent Percentage of FGA that are layups
              </option>
              <option
                value="opp_fg_layup"
              >
                Opponent Number of made layups
              </option>
              <option
                value="opp_pct_fg3a_corner"
              >
                Opponent Percentage of 3-Pt attempts from the corner
              </option>
              <option
                value="opp_fg3_pct_corner"
              >
                Opponent Corner 3-Pt Percentage
              </option>
            </select>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
          />
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
          />
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
          >
            <canvas
              height={150}
              role="img"
              width={300}
            />
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
